plugins/woocommerce/tests/api-core-tests/tests/customers/customers-crud.test.js

```diff
@@ -0,0 +1,451 @@
+			const response = await request.put(
+				`/wp-json/wc/v3/customers/1`, {
+					data: {
+						first_name: 'admin',
+						billing: {
+							first_name: 'admin'
+						},
+						shipping: {
+							first_name: 'admin'
+						}
+					}
+				}
+			);
+			const responseJSON = await response.json();
+			expect(response.status()).toEqual(200);
+			expect(responseJSON.first_name).toEqual('admin');
+			expect(responseJSON.billing.first_name).toEqual('admin');
+			expect(responseJSON.shipping.first_name).toEqual('admin');
+		});
+
+		test('retrieve after update admin', async ({
+			request
+		}) => {
+			// call API to retrieve all customers
+			const response = await request.get('/wp-json/wc/v3/customers/1');
+			const responseJSON = await response.json();
+			expect(response.status()).toEqual(200);
+			expect(responseJSON.first_name).toEqual('admin');
+			expect(responseJSON.billing.first_name).toEqual('admin');
+			expect(responseJSON.shipping.first_name).toEqual('admin');
+		});
+
+		test(`can update the subscriber user/customer`, async ({
+			request,
+		}) => {
+			// update customer names (billing and shipping) to Jane
+			// (these were initialised blank, only regular first_name was populated)
+			const response = await request.put(
+				`/wp-json/wc/v3/customers/2`, {
+					data: {
+						billing: {
+							first_name: 'Jane'
+						},
+						shipping: {
+							first_name: 'Jane'
+						}
+					}
+				}
+			);
+			const responseJSON = await response.json();
+			expect(response.status()).toEqual(200);
+			expect(responseJSON.first_name).toEqual('Jane');
+			expect(responseJSON.billing.first_name).toEqual('Jane');
+			expect(responseJSON.shipping.first_name).toEqual('Jane');
+		});
+
+		test('retrieve after update subscriber', async ({
+			request
+		}) => {
+			// call API to retrieve all customers
+			const response = await request.get('/wp-json/wc/v3/customers/2');
+			const responseJSON = await response.json();
+			expect(response.status()).toEqual(200);
+			expect(responseJSON.first_name).toEqual('Jane');
+			expect(responseJSON.billing.first_name).toEqual('Jane');
+			expect(responseJSON.shipping.first_name).toEqual('Jane');
+		});
+
+		test(`can update a customer`, async ({
+			request,
+		}) => {
+			// update customer names (regular, billing and shipping) from John to Jack
+			const response = await request.put(
+				`/wp-json/wc/v3/customers/${ customerId }`, {
+					data: {
+						first_name: 'Jack',
+						billing: {
+							first_name: 'Jack'
+						},
+						shipping: {
+							first_name: 'Jack'
+						}
+					}
+				}
+			);
+			const responseJSON = await response.json();
+			expect(response.status()).toEqual(200);
+			expect(responseJSON.first_name).toEqual('Jack');
+			expect(responseJSON.billing.first_name).toEqual('Jack');
+			expect(responseJSON.shipping.first_name).toEqual('Jack');
+		});
+
+		test('retrieve after update customer', async ({
+			request
+		}) => {
+			// call API to retrieve all customers
```

Comment:
Minor comment update:

```suggestion
			// call API to retrieve the updated customer we created above
```

Reply:
Changes look good @nigeljamesstevenson! This is the only last thing to update and I think we're good to merge once this comment is updated.